### PR TITLE
completes PR #304

### PR DESCRIPTION
**Pull request by avastar**
This completes the requested changes for PR #304 
@mbg033

### CODE REVIEW

File: src/rpc/core_rpc_server.h [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
 //    prior written permission.
-// 
+//
 // THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
 // EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
 // MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
 // THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 // SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
 // PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
```
Comment (mbg033) [inline src/rpc/core_rpc_server.h]: looks like this line inserted by mistake?

Comment (avastar) [reply src/rpc/core_rpc_server.h]: yes, ctrl+arrow key moved it and i lost where it went

**Comment by avastar** [comment]
> 
> 
> @avastar thank you. Could you please remove all the "formatting" changes (like removing extra spaces) from your commit?

fixed, had to turn off auto trailing empty space removal

**Comment by avastar** [comment]
@mbg033 ready for review

**Comment by avastar** [comment]
i have compiled and tested it. everything is working great (including any and all commits since 1.8.2).

**Comment by jagerman** [comment]
This change compared to #304 is wrong.  The stake code should either be in `on_transfer_split` or should be in *both* `on_transfer` and `on_transfer_split`, but not in just `on_transfer`.  The cli wallet doesn't require that a stake fit into a single transaction, so why should the RPC interface?

@mbg033 's comments on this issue originally were:
> `transfer_split` expects more than one destinations on input, but in case "stake_transfer" there should be one destination. It looks confusing, please move this functionality to `transfer`

I have no idea why he thought this, but it is wrong: the difference between `transfer` and `transfer_split` is that `transfer_split` allows the transfer to span multiple transactions.  BOTH functions accept multiple destinations.  `transfer_split` corresponds exactly to the way the command-line `transfer` and `locked_transfer` commands work in the cli wallet.  `transfer` is effectively deprecated for ordinary transfers unless you have a strong need to not split a transfer into multiple transactions.

Graft has no requirement for stakes to fit into a single transaction, so restricting this to a single transaction in the RPC interface makes no sense.  In other words: #304 was perfectly fine already, but if you really want to support through the seldom-used non-split transfer RPC interface, you could merge both PRs.

**Comment by avastar** [comment]
I was wondering about this myself as I've been looking into the stake lock
bug. Could transfer split be used to create two transactions, one to stake
(with the unlock_time) and the second to handle the change?

**Comment by mbg033** [comment]
> I have no idea why he thought this, but it is wrong: the difference between `transfer` and `transfer_split` is that `transfer_split` allows the transfer to span multiple transactions. BOTH functions accept multiple destinations. `transfer_split` corresponds exactly to the way the command-line `transfer` and `locked_transfer` commands work in the cli wallet. `transfer` is effectively deprecated for ordinary transfers unless you have a strong need to not split a transfer into multiple transactions.


@jagerman yes, your're right, my bad. I was mistakenly confused with the code that doesn't allow more than one destination in the `transfer_split`  call. Will merge both

**Pull request merged**